docs/DeveloperGuide.md

```diff
@@ -1,37 +1,209 @@
+
+- **State Checking**: When user inputs a command, the `Parser` queries the `State` object to check whether the system is in `MAIN_STATE` or `TASK_STATE`.
+- **State Transitions**: When a `SelectPatientCommand` is executed, the system transitions from `MAIN_STATE` to `TASK_STATE`. Vice versa for `BackCommand`.
+- **Command Execution**: Commands are executed based on the current state. For example, in `MAIN_STATE`, patient-related commands are processed, while in `TASK_STATE`, task-related commands are handled.
 
 ## Product scope
+
 ### Target user profile
 
-{Describe the target user profile}
+- Nurses who need to manage their tasks for each patient with tagging.
+- Nurses who need to keep track of their tasks for each patient.
+- Is reasonably comfortable with the command line interface.
+- Is familiar with the concept of tasks and patients.
 
 ### Value proposition
 
-{Describe the value proposition: what problem does it solve?}
+- MediTask is a handy tool for nurses to coordinate their tasks according to patients.
 
 ## User Stories
 
-|Version| As a ... | I want to ... | So that I can ...|
-|--------|----------|---------------|------------------|
-|v1.0|new user|see usage instructions|refer to them when I forget how to use the application|
-|v2.0|user|find a to-do item by name|locate a to-do without having to go through the entire list|
+| Version | As a ... | I want to ... | So that I can ... |
+|---------|----------|---------------|-------------------|
+| v1.0    | nurse    | add tasks with specific details (e.g., patient illness, precautions) | ensure all safety and medical steps are followed for each patient |
+| v1.0    | nurse    | delete tasks that are no longer relevant | keep my task list up to date and avoid unnecessary clutter |
+| v1.0    | nurse    | mark tasks as completed | stay organized and ensure all tasks are done during my shift |
+| v1.0    | nurse    | unmark tasks that were incorrectly marked as completed | quickly correct errors and keep an accurate account of ongoing tasks |
+| v1.0    | nurse    | save my tasks | access and view them after closing and reopening the interface |
+| v1.0    | nurse    | list my tasks | see all my tasks that I have currently |
+
 
 ## Non-Functional Requirements
 
-{Give non-functional requirements}
+
+1. Should work on any mainstream OS as long as it has Java 17 or above installed.
+
+2. Should be able to hold up to 1000 persons without a noticeable sluggishness in performance for typical usage.
+
+3. A user with above average typing speed for regular English text (i.e. not code, not system admin commands) should be able to accomplish most of the tasks faster using commands than using the mouse.
 
 ## Glossary
 
-* *glossary item* - Definition
+- **Mainstream OS**: Windows, Linux, Unix, MacOS
+- **Personal data**: Data that can be used to identify a person, such as a name and medical records.
```

Comment:
Good way to ensure everyone is on the same page in what "personal data" refers to. Good job!